---
title: "Protist diversity and biogeography in the Okinawa Trough"
author: "Maggi Brisbin"
date: "4/16/2019"
output:
  html_document:
    number_sections: yes
    theme: cerulean
    toc: yes
    toc_depth: 6
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '6'
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(cache=TRUE, fig.path='figures/', fig.width=8, fig.height=6.5 )
```
#Map of sampling stations in the Okinawa Trough 

Load required package:
```{r, warning=FALSE, message = FALSE}
library(marmap)
```

Import bathymetry data from NOAA:
```
b = getNOAA.bathy(lon1 = 115, lon2 = 140, lat1 = 35, lat2 = 20, resolution = 1)
```

Import dataframe with the latitude and longitude of sampling points from CTD (in decimal degrees):
```{r}
pts <- read.csv("Mirai_CTD_lat_long.csv")
pts$Station<- as.character(pts$Station)
str(pts)
```

Make dataframe with map labels and their lat/long: 
```{r}
Clon = c(127.5,131.25, 121.6, 121.03, 120.8, 126)
Clat= c(26, 31.93, 31.1, 23, 27.2, 22)
CLabels= c("Okinawa", "Japan", "China", "Taiwan", "East China Sea", "Philippine Sea")
Countries = cbind.data.frame(Clon, Clat, CLabels)
str(Countries)
```

Make a dataframe with vent sites:
```{r}
vents <- read.csv("VentSites.csv")
str(vents)
```

Make plot one layer at a time: 
```
#tiff("MiraiFiltersMap.tiff", height= 8, width =8, units = 'in', res=300) # uncomment to save plot to file
plot(b, image = TRUE, land = TRUE, xlim=c(120,135), ylim=c(22, 32), lwd = 0.03, bpal = list(c(0, max(b), grey(.7), grey(.9), grey(.95)),
c(min(b), 0, "darkblue", "lightblue"))) 
#plot(b, image = TRUE, land = TRUE, xlim=c(120,135), ylim=c(22, 32), lwd = 0.03, bpal = Newblues(100), cex.lab=1.5, cex.axis=1.5, cex.main=1.5, cex.sub=1.5)
#bathymetry colors
plot(b, lwd = 0.8, deep = 0, shallow = 0, step = 0, add = TRUE) # highlight coastline with solid black line
plot(b, deep=-200, shallow=-200, lwd = 0.4, drawlabels=T, add=T)  # Add -200m isobath
plot(b, deep=-1000, shallow=-1000, lwd = 0.4, drawlabels=T, add=T)  # Add -1000m isobath
plot(b, deep=-2000, shallow=-2000, lwd = 0.4, drawlabels=T, add=T)  # Add -2000m isobath
plot(b, deep=-3000, shallow=-3000, lwd = 0.4, drawlabels=T, add=T)  # Add -3000m isobath
points(pts, pch = 16, col = 'red') # Add sampling points
text(pts, labels=pts$Station, adj= 1.5, cex=1.2, size = 4) # Add sampling station numbers
text(Countries, labels=Countries$CLabels, cex= 1.7, adj = -0.2) # Add map labels 
points(vents, pch = 17, col = 'blue', size = 3)
#dev.off() #uncomment to save plot to file
```
![](MiraiFiltersMap_noRyukyuCurrent.png)

#Sequence analysis

## Identify and Count Amplicon Sequence Variants (ASVs) 
**Sample collection, DNA extraction, PCR, & sequencing**

Seawater collected from the subsurface chlorophyll maximum (SCM), mid-water column, and near-bottom was collected in 10 L Niskin bottles at each sampling station and surface seawater was collected by bucket. Five Liter replicates from each depth (4.5 from surface) were sequentially filtered through 10.0 and 0.2 um pore-size Polytetrafluoroethylene (PTFE) filters. Filters were flash frozen in liquid nitrogen and stored at -80 C until DNA extraction. DNA was extracted following the manufacturers protocols for the Qiagen/MoBio DNeasy PowerWater Kit, including the optional heating step to lyse cells. Amplicon PCR and sequencing library preparation followed the procedures described in the Illumina 16S Metagenomic Sequencing Library Preparation manual modified to include universal eukaryotic primers for the v4 region of the 18S rRNA gene (F: Stoeck et al. 2010, R: Brisbin et al. 2018) and amplicon PCR conditions most appropriate for these primers (annealing at 58C). The 220 samples were separated into 4 sequencing runs on the Illumina MiSeq sequencing platform with v3 chemistry for 2x300-bp sequencing. 

**Bioinformatic processing with QIIME 2**

Demultiplexed paired-end sequences provided by the OIST sequencing center were imported to QIIME 2 (v2018.11) software (Bolyen et al. 2019). The Divisive Amplicon Denoising Algorithm (DADA) was implemented with the DADA2 plug-in for Qiime2 for quality filtering, chimera removal, and feature table construction (Callahan et al. 2015) separately for each sequencing run before merging the four resulting feature tables. Taxonomy was assigned to Amplicon Sequence Variants (ASVs) in the feature table by training a classifier on the Protist Ribosomal Reference (PR2) database to classify ASVs (Guillo et al. 2012). The feature table and assigned taxonomy were then exported from Qiime2 for further analysis. 

**Below is an example Qiime2 workflow for 1 sequencing run:**

Activate an anaconda environment for qiime2:
`source activate qiime2-2018.11`

Import sequencing data (all sequences for eacg sequencing run must be in their own directory):
```
qiime tools import --type 'SampleData[PairedEndSequencesWithQuality]' --input-path ./seqs/seqrun1/ --input-format CasavaOneEightSingleLanePerSampleDirFmt --output-path run1_demux-paired-end.qza
```

Visualize squencing results (use these graphs to decide what length to trim sequences)
```
qiime demux summarize --i-data run1_demux-paired-end.qza --o-visualization run1_demux.qzv
```

Run the DADA2 algorithm: 
```
qiime dada2 denoise-paired --i-demultiplexed-seqs run1_demux-paired-end.qza --output-dir ./dd2run1 --o-representative-sequences run1_rep-seqs --p-trim-left-f 10 --p-trim-left-r 10 --p-trunc-len-f 260 --p-trunc-len-r 250 --p-n-threads 3
```

Check results:
```
qiime feature-table summarize --i-table ./dd2run1/table.qza --o-visualization ./dd2run1/table.qzv --m-sample-metadata-file ~/desktop/MiraiFilters/sampleMaptxt.txt 
```  

```
qiime metadata tabulate --m-input-file dd2run1/denoising_stats.qza --o-visualization dd2run1/denoising_stats.qzv
```

Merge feature tables from multiple sequencing runs: 
```
qiime feature-table merge --i-tables ./dd2run1/table.qza --i-tables ./dd2run2/table.qza --i-tables ./dd2run3/table.qza --i-tables ./dd2run4/table.qza --o-merged-table mergedtable.qza
```

```
qiime feature-table merge-seqs --i-data run1_rep-seqs.qza --i-data run2_rep-seqs.qza --i-data run3_rep-seqs.qza --i-data run4_rep-seqs.qza --o-merged-data merged-rep-seqs.qza
```

Import taxonomy database: 
```
qiime tools import --type 'FeatureData[Sequence]' --input-path pr2_version_4.11.1_mothur.fasta --output-path pr2.qza
```

```
qiime tools import --type 'FeatureData[Taxonomy]' --input-format HeaderlessTSVTaxonomyFormat --input-path pr2_version_4.11.1_mothur.tax -output-path pr2_tax.qza
```

Select V4 region from PR2 sequences: 
```
qiime feature-classifier extract-reads --i-sequences pr2.qza --p-f-primer CCAGCASCYGCGGTAATTCC --p-r-primer ACTTTCGTTCTTGATYRA --o-reads pr2_v4.qza
```

Train the classifier: 
```
qiime feature-classifier fit-classifier-naive-bayes --i-reference-reads pr2_v4.qza --i-reference-taxonomy pr2_tax.qza --o-classifier pr2_v4_classifier.qza
```

Classify: 
```
qiime feature-classifier classify-sklearn --i-classifier pr2_v4_classifier.qza --i-reads merged-rep-seqs.qza --o-classification taxonomy.qza
```

```
qiime metadata tabulate --m-input-file taxonomy.qza --o-visualization taxonomy.qzv
```

Export the .tsv taxonomy file from `taxonomy.qzv` for use in following steps. 

##Load data into R & prepare phyloseq objects

Load packages:
```{r, message=FALSE, warning=FALSE}
library(tidyverse)
library(qiime2R)
library('phyloseq')
library('vegan')
library('ggplot2')
library(RColorBrewer)
library(DESeq2)
library(reshape)
library("wesanderson")
library("gridExtra")
library("metagMisc")
library("wesanderson")
library("breakaway")
library("CoDaSeq")
library("ggbiplot")
library(ggpubr)
```

Set default colors:
```{r}
ap<- c("#F1B6A1", "#D4A52A", "#E3E5DB", "#A5CFCC", "#0E899F", "#A83860", "#ED91BC", "#DB5339", "#F58851", "#42465C", "#1E479A", "#F7CDA4", "#CF529C", "#11638C")
```

Convert QIIME 2 artifacts to phyloseq objects: 
```{r, message = FALSE, warning = FALSE}
phyloseq<-qza_to_phyloseq(features="mergedtable.qza")
```

Import sample data:
```{r}
metatable <- read.csv("sampleMap.csv")
row.names(metatable) <- metatable[[1]]

metatable$Station <- as.character(metatable$Station)
metatable$filter <- as.character(metatable$filter)
metatable$VentANAHTM<-as.factor(metatable$VentANAHTM)

META<- sample_data(metatable)
str(META)
```

Load PR2 taxonomy:
```{r}
taxonomy <- read.csv("taxonomy.csv", stringsAsFactors = FALSE)
names(taxonomy) <- c("row", "tax", "Confidence")
row.names(taxonomy) <-taxonomy[[1]]
taxonomy <- taxonomy[,(-1)]
taxonomy <-  separate(taxonomy, tax, c("D0","D1", "D2", "D3", "D4", "D5", "D6", "D7", "D8", "D9", "D10", "D11", "D12", "D13", "D14"), sep = ";", fill = "right")
taxonomy <- taxonomy[,c(1:8)]
taxmat <- as.matrix(taxonomy)
TAX = tax_table(taxmat)
str(TAX)
```

Add taxonomy to phyloseq object:
```{r}
ps = merge_phyloseq(phyloseq, TAX, META)
```

##Prevalence Plots 

Prevalence plots display the total abundance of an ASV in the full data set v. the fraction of total samples that ASV is found in. 

```{r, cache = TRUE}
prevdf = apply(X = otu_table(ps),
               MARGIN = ifelse(taxa_are_rows(ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})

prevdf = data.frame(Prevalence = prevdf,
                    TotalAbundance = taxa_sums(ps),
                    tax_table(ps))

#plyr::ddply(prevdf, "D2", function(df1){cbind(mean(df1$Prevalence),sum(df1$Prevalence))})
```

Prevalence plot:
```{r, warning = FALSE, cache = TRUE}
prevplot1<-ggplot(prevdf, aes(TotalAbundance, Prevalence / nsamples(ps),color=D2)) +  geom_point(size = 2, alpha = 0.7) + 
  theme_bw()+
  scale_x_log10() +  xlab("Total Abundance") + ylab("Prevalence [Frac. Samples]") +
  facet_wrap(~D1) + theme(legend.position="none")

prevplot1
```

*We don't apply prevalence filter* because we are interested in rare ASVs in deep water masses and at hydrothermal vents. 

##Basic statistics

```{r}
OTUs <- data.frame(otu_table(ps))
```

Total number of ASVs in the data set:
```{r}
OTUsRS<- OTUs
OTUsRS$RowSum <- rowSums(OTUsRS)
OTUsRSnoZero <- OTUsRS$RowSum!=0
sum(OTUsRSnoZero)
```

Total number of ASVs per sample (range and mean):

```{r}
OTUs0 <- OTUs!=0 #is this number not a zero? true (1) or false (0)
csums <- colSums(OTUs0) # col sums = observed ASV richness
csumdf <- as.data.frame(csums)
max(csumdf$csums) #1906
min(csumdf$csums) #49
mean(csumdf$csums) #730
```

##Alpha Diversity

**Breakaway Modeled Richness for all samples by depth layer:** 
```{r , warning = FALSE, message = FALSE}
ba <- breakaway(ps)

badf<- summary(ba) %>%
  add_column("SampleID" = ps %>% otu_table %>% sample_names)

badf<- merge(badf, metatable, by = "SampleID")

baPlot <- ggplot(badf, aes(x=Depth, y=estimate, fill=filter)) + geom_boxplot() + theme_bw() + scale_fill_manual(values = c("#E3E5DB", "#11638C"), labels=c("10.0", "0.2")) + theme(text = element_text(size=14)) +ylab("Richness Estimate")

baPlot
ggsave("breakaway_4depths.tiff", width = 6, height = 4)
```

**Breakaway alpha diversity significance testing for all samples by depth layer:**
```{r}
bt <- betta(summary(ba)$estimate,
            summary(ba)$error,
            make_design_matrix(ps, "Depth"))
bt$table
```
The modeled richness in the SCM is significantly higher than all other sample types, which are not significantly different from each other. 

Breakaway Modeled Richness for all bottom water samples by sampling station: 
```{r}
bottomraw<- subset_samples(ps, Depth=="Bottom")
```

```{r richness_all_bottoms, warning = FALSE, message  = FALSE}
bba <- breakaway(bottomraw)

bbadf<- summary(bba) %>%
  add_column("SampleID" = bottomraw %>% otu_table %>% sample_names)

bbadf<- merge(bbadf, metatable, by = "SampleID")
bbadf$Station_f <- factor(bbadf$Station, levels=c("11", "10", "9", "8", "3", "4", "2", "5", "12", "13", "14", "15", "17", "18"))
bbaPlot <- ggplot(bbadf, aes(x=Station_f, y=estimate, fill=filter)) + geom_boxplot() + theme_bw() + scale_fill_manual(values = c("#E3E5DB", "#11638C"), labels=c("10.0", "0.2")) + theme(text = element_text(size=14)) +ylab("Richness Estimate") +xlab("Station")

bbaPlot
ggsave("breakaway_bottom.tiff", width = 6, height = 4)
```

Breakaway alpha diversity significance testing for bottom samples by vent (stations 2, 10), plume (station 11), or no hydrothermal activity (all other stations):
```{r}
bt <- betta(summary(bba)$estimate,
            summary(bba)$error,
            make_design_matrix(bottomraw, "VentPlume"))
bt$table
```
Vent/Plume sites are not more or less diverse than sites without hydrothermal influence.

##Taxonomic Filtering

* Remove ASV's without a D1 ("Kingdom") taxonomy assignment 
* Remove ASV's assigned as Metazoa at D2

```{r}
#table(tax_table(psN)[, "D1"], exclude = NULL)
ps<-subset_taxa(ps, D1 != "")
ps<-subset_taxa(ps, D1 != "NA")
ps<-subset_taxa(ps, D2 != "Metazoa")
```

```{r}
OTUs <- data.frame(otu_table(ps))
```

Check ASV richness to see how much it changes after filtering:
```{r}
OTUs0 <- OTUs!=0
csums <- colSums(OTUs0)
csumdf <- as.data.frame(csums)
max(csumdf$csums) #1800
min(csumdf$csums) #45
mean(csumdf$csums) #685
```

##Distance and Ordination 

###All Samples

####Atchison distance (CLR + Euclidean)
* compute CLR normalization, CLR = centered log-ratio, log(x/gx) where gx is the geomentric mean of vector x
* Then use Euclidean distance 
* PCoA and/or PCA
* PERMANOVA 
```{r, message = FALSE, warning = FALSE}
OTU4clr<- data.frame(t(data.frame(otu_table(ps))))
row.names(OTU4clr) <- gsub("\\.", "", row.names(OTU4clr))
OTUs.clr <- codaSeq.clr(OTU4clr + 0.5, samples.by.row=TRUE)
OTU2 <- otu_table(as.matrix(OTUs.clr), taxa_are_rows = FALSE)

row.names(metatable) <- gsub("-", "", row.names(metatable))
META2<- sample_data(metatable)

psCLR <- phyloseq(OTU2,TAX,META2)
```

```{r AitchinsonPCoA_All, warning = FALSE, message = FALSE }
ordu = ordinate(psCLR, "PCoA", "euclidean")
p<-plot_ordination(psCLR, ordu, color="Depth", shape="filter")+theme_bw() +scale_color_manual(values=c(ap[8], ap[12], ap[5], ap[4])) + scale_shape_discrete(labels=c("10.0", "0.2")) +geom_point(size=3) +theme(text = element_text(size=16)) 
p
```
```{r}
#ggsave("pcoa_allsamples.tiff", width = 6, height = 4)
```

###Ordination for sample subgroups (Depth and Depth/Filter-type)

The strongest determinant of clustering seems to be depth. To better observe clustering within depth groups, subset by depth and then filter type.

####Surface

```{r pcoa_surf}
physeqSurf<- subset_samples(psCLR, Depth == "Surface")
ordu = ordinate(physeqSurf, "PCoA", "euclidean")
pSurf<-plot_ordination(physeqSurf, ordu, color="Depth", shape="filter")+theme_bw() +scale_color_manual(values= c(ap[4]))+ geom_point(size=3) +ggtitle("Surface") +theme(text = element_text(size=16)) + guides(color = FALSE) + scale_shape_discrete(labels=c("10.0", "0.2"))
pSurf
```
Samples cluster by filter type. 

####Surface 10.0

**PCA**

```{r surf10_pcaLines}
physeqSurf<- subset_samples(psCLR, Depth == "Surface" & filter == "10")
OTUsSurf10 <- data.frame(otu_table(physeqSurf))
meta <- metatable[row.names(metatable) %in% row.names(OTUsSurf10),]
Surface_10<- prcomp(OTUsSurf10)
plot(Surface_10, type="lines")
```

```{r PCA_Surf10}
PCA_Surf10<-ggbiplot(Surface_10, var.axes = FALSE, groups= meta$Region_SKN, ellipse=TRUE) +theme_bw() +scale_color_manual(values=c(ap[4], ap[11], ap[2]))+ theme(text = element_text(size=14)) +  theme(legend.position="bottom") +ggtitle("Surface 10.0")
PCA_Surf10
```

**PERMANOVA**
```{r}
set.seed(1)
adonis(vegdist(OTUsSurf10, method = "euclidean") ~ Region_SKN, data = meta)
```

**Pairwise PERMANOVA**
```{r}
tst<-adonis_pairwise(x=meta, dd=vegdist(OTUsSurf10, method = "euclidean"), group.var="Region_SKN")
tst$Adonis.tab
```
* KG to North - Diff
* KG to South - NOT diff
* North to South - Diff

####Surface 0.2

**PCA**
```{r}
physeqSurf2<- subset_samples(psCLR, Depth == "Surface" & filter == "2")
```
```{r surf2_pcaLines}
#tiff("variance_surface2.tiff", height= 8, width =8, units = 'in', res=70) # uncomment to save plot to file
OTUsSurf2 <- data.frame(otu_table(physeqSurf2))
meta <- metatable[row.names(metatable) %in% row.names(OTUsSurf2),]
surface2 <- prcomp(OTUsSurf2)
plot(surface2, type="lines")
#dev.off()
```

```{r PCA_Surf2}
PCA_Surf2<-ggbiplot(surface2, var.axes = FALSE, groups= meta$Region_SKN, ellipse=TRUE ) +theme_bw() +scale_color_manual(values=c(ap[4], ap[11], ap[2]))+ theme(text = element_text(size=14)) +  theme(legend.position="bottom") +ggtitle("Surface 0.2")
PCA_Surf2
```

**PERMANOVA**
```{r}
set.seed(1)
adonis(vegdist(OTUsSurf2, method = "euclidean") ~ Region_SKN, data = meta)
```

**Pairwise PERMANOVA**
```{r}
tst<-adonis_pairwise(x=meta, dd=vegdist(OTUsSurf2, method = "euclidean"), group.var="Region_SKN")
tst$Adonis.tab
```

* KG to North - Diff
* KG to South - NOT diff
* North to South - Diff

###SCM
```{r PCoA_SCM }
physeqSCM<- subset_samples(psCLR, Depth == "SCM")
ordu = ordinate(physeqSCM, "PCoA", "euclidean")
pSCM<-plot_ordination(physeqSCM, ordu, color="Depth", shape="filter")+theme_bw() +scale_color_manual(values=c(ap[5]))+ geom_point(size=3) +ggtitle("SCM") +theme(text = element_text(size=16))
pSCM
```
Again, strong clustering by filter type. 

####SCM 10.0

**PCA**
```{r SCM10_pcaLines}
physeqSCM10<- subset_samples(psCLR, Depth == "SCM" & filter =="10")
OTUsSCM10 <- data.frame(otu_table(physeqSCM10))
meta <- metatable[row.names(metatable) %in% row.names(OTUsSCM10),]
SCM10 <- prcomp(OTUsSCM10)
plot(SCM10, type="lines")
```

```{r PCA_SCM10}
PCA_SCM10<-ggbiplot(SCM10, var.axes = FALSE, groups= meta$Region_SKN, ellipse=TRUE ) +theme_bw() +scale_color_manual(values=c(ap[4], ap[11], ap[2]))+ theme(text = element_text(size=14)) +  theme(legend.position="bottom") +ggtitle("SCM 10.0")
PCA_SCM10
```

**PERMANOVA**
```{r}
set.seed(1)
adonis(vegdist(OTUsSCM10, method = "euclidean") ~ Region_SKN, data = meta)
```
**pairwise PERMANOVA**

```{r}
tst<-adonis_pairwise(x=meta, dd=vegdist(OTUsSCM10, method = "euclidean"), group.var="Region_SKN")
tst$Adonis.tab
```
* KG and NOT = Different
* NOT and SOT = Different
* KG and SOT = Not Different

####SCM 0.2

**PCA**
```{r SCM2_pcaLines}
physeqSCM2<- subset_samples(psCLR, Depth == "SCM" & filter == "2")
OTUsSCM2 <- data.frame(otu_table(physeqSCM2))
meta <- metatable[row.names(metatable) %in% row.names(OTUsSCM2),]
SCM2 <- prcomp(OTUsSCM2)
plot(SCM2, type="lines")
```

```{r PCA_SCM2}
PCA_SCM2<-ggbiplot(SCM2, var.axes = FALSE, groups= meta$Region_SKN, ellipse=TRUE ) +theme_bw() +scale_color_manual(values=c(ap[4], ap[11], ap[2]))+ theme(text = element_text(size=14)) +  theme(legend.position="bottom") +ggtitle("SCM 0.2")
PCA_SCM2
```

**PERMANOVA**
```{r}
set.seed(1)
adonis(vegdist(OTUsSCM2, method = "euclidean") ~ Region_SKN, data = meta)
```

###Mid

```{r PCoA_Mid}
physeqMid<- subset_samples(psCLR, Depth == "Mid")
ordu = ordinate(physeqMid, "PCoA", "euclidean")
pMid<-plot_ordination(physeqMid, ordu, color="Depth", shape = "filter")+theme_bw() +scale_color_manual(values=ap[12])+ geom_point(size=3) +ggtitle("Mid")+theme(text = element_text(size=16))
pMid
```

Samples still cluster by filter pore-size, but now on the secondary axis.

####Mid 10.0

**PCA**
```{r Mid10_pcaLines}
physeqMid10<- subset_samples(psCLR, Depth == "Mid" & filter == "10")
OTUsMid10 <- data.frame(otu_table(physeqMid10))
meta <- metatable[row.names(metatable) %in% row.names(OTUsMid10),]
Mid10 <- prcomp(OTUsMid10)
plot(Mid10, type="lines")
```

```{r PCA_Mid10}
PCA_Mid10<-ggbiplot(Mid10, var.axes = FALSE, groups= meta$Region_SKN, ellipse=TRUE ) +theme_bw() +scale_color_manual(values=c(ap[4], ap[11], ap[2]))+ theme(text = element_text(size=14)) +  theme(legend.position="bottom") +ggtitle("Mid 10.0")
PCA_Mid10
```

**PERMANOVA by Region**
```{r}
adonis(vegdist(OTUsMid10, method = "euclidean") ~ Region_SKN, data = meta, permutations = 999)
```
*not significant at our cut-off*

####Mid 0.2

**PCA**
```{r Mid2}
physeqMid2<- subset_samples(psCLR, Depth == "Mid" & filter == "2")
OTUsMid2 <- data.frame(otu_table(physeqMid2))
meta <- metatable[row.names(metatable) %in% row.names(OTUsMid2),]
Mid2 <- prcomp(OTUsMid2)
plot(Mid2, type="lines")
```

```{r PCA_Mid2}
PCA_Mid2<-ggbiplot(Mid2, var.axes = FALSE, groups= meta$Region_SKN, ellipse=TRUE ) +theme_bw() +scale_color_manual(values=c(ap[4], ap[11], ap[2]))+ theme(text = element_text(size=14)) +  theme(legend.position="bottom") +ggtitle("Mid 0.2")
PCA_Mid2
```

```{r}
#ggsave("PCA_Mid2.png")
```

**PERMANOVA by Region**
```{r}
set.seed(1)
adonis(vegdist(OTUsMid2, method = "euclidean") ~ Region_SKN, data = meta)
```

**pairwise PERMANOVA**
```{r}
tst<-adonis_pairwise(x=meta, dd=vegdist(OTUsMid2, method = "euclidean"), group.var="Region_SKN")
tst$Adonis.tab
```

* North to South = diff
* KG to North = diff
* KG to South = not diff

###Bottom

```{r PCoA_Bottom}
physeqBottom<- subset_samples(psCLR, Depth == "Bottom")
ordu = ordinate(physeqBottom, "PCoA", "euclidean")
pBottom<-plot_ordination(physeqBottom, ordu, color="Depth", shape = "filter")+theme_bw() +scale_color_manual(values=ap[8])+ geom_point(size=3) +ggtitle("Bottom")+theme(text = element_text(size=16))
pBottom 
```
No clustering by filter type. 

For consistency, split by filter type for the NOT/SOT/KG biogeography: 

####Bottom 10.0

**PCA**
```{r bottom10_pcaLines}
physeqBottom10<- subset_samples(psCLR, Depth == "Bottom" & filter == "10")
OTUsBottom10 <- data.frame(otu_table(physeqBottom10))
meta <- metatable[row.names(metatable) %in% row.names(OTUsBottom10),]
Bottom10 <- prcomp(OTUsBottom10)
plot(Bottom10, type="lines")
```

```{r PCA_Bottom10}
PCA_Bottom10<-ggbiplot(Bottom10, var.axes = FALSE, groups= meta$Region_SKN, ellipse=TRUE ) +theme_bw() +scale_color_manual(values=c(ap[4], ap[11], ap[2]))+ theme(text = element_text(size=14)) +  theme(legend.position="bottom") +ggtitle("Bottom 10.0")
PCA_Bottom10
```

**PERMANOVA by Region**
```{r}
set.seed(1)
adonis(vegdist(OTUsBottom10, method = "euclidean") ~ Region_SKN , data = meta)
```
*not significant at our cut-off*


####Bottom 0.2

**PCA**
```{r}
physeqBottom2<- subset_samples(psCLR, Depth == "Bottom" & filter == "2")
OTUsBottom2 <- data.frame(otu_table(physeqBottom2))
meta <- metatable[row.names(metatable) %in% row.names(OTUsBottom2),]
Bottom2 <- prcomp(OTUsBottom2)
plot(Bottom2, type="lines")
```

```{r PCA_Bottom2}
PCA_Bottom2<-ggbiplot(Bottom2, var.axes = FALSE, groups= meta$Region_SKN, ellipse=TRUE ) +theme_bw() +scale_color_manual(values=c(ap[4], ap[11], ap[2]))+ theme(text = element_text(size=14)) +  theme(legend.position="bottom") +ggtitle("Bottom 0.2")
PCA_Bottom2
```

**PERMANOVA by region**
```{r}
set.seed(1)
adonis(vegdist(OTUsBottom2, method = "euclidean") ~ Region_SKN, data = meta)
```

**Pairwise PERMANOVA**
```{r}
tst<-adonis_pairwise(x=meta, dd=vegdist(OTUsBottom2, method = "euclidean"), group.var="Region_SKN")
tst$Adonis.tab
```

* North to South = diff
* KG to North = not diff
* KG to South =  diff

*Pattern reversed from that seen in Mid, SCM, and Surface*

**Supplemental Figure 1**
```{r suppFig1, warning = FALSE, message = FALSE}
supp1<-ggarrange(pSurf, pSCM, pMid, pBottom, common.legend = TRUE, legend = "bottom")
supp1
```
```{r}
#ggsave("Supplemental1.png", width = 6, height = 6 )
```

**Bottom-water by Hydrothermal Activity**

For Vent v. Plume - use 10.0 and 0.2 samples together because 0.2 and 10.0 pore-size filters do not cluster separately in the PCoA and necessary because there are a low number of vent sites compared to non-vent sites 

**PERMANOVA**

```{r}
set.seed(1)
physeqBottom<- subset_samples(psCLR, Depth == "Bottom")
OTUsBottom <- data.frame(otu_table(physeqBottom))
meta <- metatable[row.names(metatable) %in% row.names(OTUsBottom),]
adonis(vegdist(OTUsBottom, method = "euclidean") ~ VentPlume, data = meta)
```
**pairwise PERMANOVA**
```{r}
tst<-adonis_pairwise(x=meta, dd=vegdist(OTUsBottom, method = "euclidean"), group.var="VentPlume")
tst$Adonis.tab
```

* Vents v. Plume = Not diff
* Vents v. NO hydrothermal activity = Different
* Plume v. NO hydrothermal activity = Different

**PCA**
```{r PRC_BottomVents}
Bottom <- prcomp(OTUsBottom)
plot(Bottom, type="lines")
```

```{r PCA_Bottom}
PCA_Bottom<-ggbiplot(Bottom, var.axes = FALSE, groups= meta$VentPlume, ellipse=TRUE ) +theme_bw() +scale_color_manual(values= c(wes_palette("Cavalcanti1")[4], wes_palette("Cavalcanti1")[1], wes_palette("Cavalcanti1")[5] ), labels= c("none", "plume", "vent"))+ theme(text = element_text(size=16)) +  theme(legend.position="bottom")
PCA_Bottom
```

## Relative Abundance Plots

Merge Samples --> collapse replicates (for Relative Abundance Plots)

Prepare metadata table:
```{r}
metatable <- read.csv("sampleMap.csv")
row.names(metatable) <- metatable[[1]]
metatable$desc <- paste(metatable$Station, metatable$Depth, metatable$filter, sep="-")
metatable$Depth <- as.character(metatable$Depth)
metatable$Station <-as.character(metatable$Station)
metatable$filter<-as.character(metatable$filter)
metatable<- metatable[,-which(names(metatable) == "SampleID")]
META<- sample_data(metatable)
ps2<- ps
sample_data(ps2) <- META
```

```{r}
print(ps2)
```
```{r}
str(sample_data(ps2))
```

Merge:
```{r, warning=FALSE}
mergedps <- merge_samples(ps2, "desc")
print(mergedps)
```
*number of samples reduced from 211 to 112*

```{r}
sample_names(mergedps)
```

But metatable inside phyloseq object was disturbed ... 
```{r}
str(sample_data(mergedps))
```

Fix meta table in phyloseq object: 
```{r}
meta2<-as.data.frame(sample_data(mergedps))
split<- do.call(rbind, strsplit(row.names(meta2), "-"))
meta2$Depth <- split[,2]
meta2$filter<-split[,3]
meta2$desc <- row.names(meta2)
meta2$Station <- as.character(meta2$Station)
meta2$Depth_f <- factor(meta2$Depth, levels=c('Surface','SCM','Mid','Bottom'))
meta2$Station_f <- factor(meta2$Station, levels=c("11", "10", "9", "8", "3", "4", "2", "5", "12", "13", "14", "15", "17", "18"))
META <-sample_data(meta2)
sample_data(mergedps)<-META
```

Fixed!
```{r}
str(sample_data(mergedps))
```

Taxa glom - for clean RA plots (so that there is a not a line in the plot for each ASV)
```{r}
mergedps<- subset_taxa(mergedps, D2 != "Metazoa")
mergedps<-subset_taxa(mergedps, D1 != "")
mergedps<-subset_taxa(mergedps, D1 != "NA")

mergedps = tax_glom(mergedps, "D2")
```

```{r}
'%ni%' <- Negate('%in%')
LowPrev<- c("Alveolata_X", "Amoebozoa_X", "Apicomplexa", "Apusomonadidae", "Conosa", "Discoba", "Eukaryota_XX", "Foraminifera", "Hilomonadea","Fungi", "Katablepharidophyta", "Lobosa", "Mesomycetozoa", "Metamonada", "Metazoa", "Opisthokonta_X", "Perkinsea", "Streptophyta", "Rhodophyta","Telonemia", "")
mergedHighPrev<- subset_taxa(mergedps, D2 %ni% LowPrev)
```

```{r}
mergedRAhp <- transform_sample_counts(mergedHighPrev, function(OTU) 100* OTU/sum(OTU))

ap2<- c("#F7CDA4", "#DB5339", "#E3E5DB", "#A5CFCC", "#11638C", "#A83860","#D4A52A", "#CF529C", "#1E479A", "#F1B6A1", "#42465C",  "#0E899F")


taxabarplot<-plot_bar(mergedRAhp, x= "Station_f", fill = "D2", facet_grid=filter~Depth_f) + scale_y_continuous(expand = c(0, 0)) + ggtitle("") + scale_fill_manual(values= ap2) + theme(legend.title=element_blank()) + geom_bar(aes(fill=D2), stat="identity", position="stack", width =0.9) +theme_classic() + theme(text = element_text(size=14))+theme(axis.text.x = element_text(angle = 90)) + xlab("Station") +ylab("Relative Abundance(%)")

taxaplot_nolegend <- taxabarplot + theme(legend.position="none")

taxaplot_nolegend

```

```{r}
#ggsave("taxaplot_nolegend_newcolors.pdf", width = 8, height = 5)
```


```{r, echo = FALSE}
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  legend
}
legend <- g_legend(taxabarplot)
plegend <- grid.arrange(legend)
```


```{r}
taxabarplotD1<-plot_bar(mergedRAhp, x= "Station_f", fill = "D1", facet_grid=filter~Depth_f) + scale_y_continuous(expand = c(0, 0)) + ggtitle("") + scale_fill_manual(values=ap) + theme(legend.title=element_blank()) + geom_bar(aes(fill=D1), stat="identity", position="stack", width =0.9) +theme_classic() + theme(text = element_text(size=14))+theme(axis.text.x = element_text(angle = 90)) + xlab("Station") +ylab("Relative Abundance(%)")

taxaplotD1_nolegend <- taxabarplotD1 + theme(legend.position="none")

taxaplotD1_nolegend
```

```{r, echo = FALSE}
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  legend
}
legend <- g_legend(taxabarplotD1)
D1legend <- grid.arrange(legend)
```

```{r BottomTaxaplotD2}
BottomTaxa<-subset_samples(mergedRAhp, Depth == "Bottom")

Bottombarplot<-plot_bar(BottomTaxa, x= "Station_f", fill = "D2", facet_grid=~filter) + scale_y_continuous(expand = c(0, 0)) + ggtitle("") + scale_fill_manual(values=ap2) + theme(legend.title=element_blank()) + geom_bar(aes(fill=D2), stat="identity", position="stack", width =0.9) +theme_classic() + theme(text = element_text(size=14))+theme(axis.text.x = element_text(angle = 90)) + xlab("Station") +ylab("Relative Abundance(%)")

Bottombarplot_nolegend <- Bottombarplot+ theme(legend.position="none")

Bottombarplot_nolegend
```

```{r}
#ggsave("Bottombarplot_nolegend.pdf", width=8, height =4)
```

##DESeq2 -- Differential abundance testing 
subset rawcount phyloseq object to just hold bottom samples:
```{r}
bottomraw<- subset_samples(ps, Depth=="Bottom")
```
First, with only station 2 and 10 considered as hydrothermal sites:
```{r}
ddse <- phyloseq_to_deseq2(bottomraw, ~VentANAHTM)
```
```{r, warning = FALSE, message = FALSE}
ddse2 <- DESeq(ddse, test="Wald", fitType="parametric")
```
```{r}
res<- results(ddse2,  cooksCutoff = FALSE)
alpha = 0.01
sigtab = res[which(res$padj < alpha), ]
dim(sigtab)
```

And if station 11 is included ...
```{r}
ddse <- phyloseq_to_deseq2(bottomraw, ~Vent)
```
```{r, warning = FALSE, message = FALSE}
ddse2 <- DESeq(ddse, test="Wald", fitType="parametric")
```

```{r, warning=FALSE, message= FALSE}
res<- results(ddse2,  cooksCutoff = FALSE)
alpha = 0.01
sigtab = res[which(res$padj < alpha), ]
dim(sigtab)
```


```{r}
sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(bottomraw)[rownames(sigtab), ], "matrix"))
sigtab = sigtab[sigtab$D2 != "Fungi",] # this fungus classification seems to be a mistake (based on original publication) and would otherwise only be classified as Eukaryota and, therefore, filtered from our data set at the very beginning of the analysis
```

```{r, warning = FALSE}
x = tapply(sigtab$log2FoldChange, sigtab$D2, function(x) max(x))
x = sort(x, TRUE)
sigtab$D2 = factor(as.character(sigtab$D2), levels=names(x))
#D3 level
x = tapply(sigtab$log2FoldChange, sigtab$D3, function(x) max(x))
x = sort(x, TRUE)
sigtab$D3 = factor(as.character(sigtab$D3), levels=names(x))
# D4 level
x = tapply(sigtab$log2FoldChange, sigtab$D4, function(x) max(x))
x = sort(x, TRUE)
sigtab$D4 = factor(as.character(sigtab$D4), levels=names(x))
#plot F0
sigplot<- ggplot(sigtab, aes(x=D2, y=log2FoldChange, color = D3)) + geom_point(size=5, alpha = 0.7) + theme(legend.title=element_blank()) + theme_bw() +
   ggtitle(" ") +
  theme(axis.text.x = element_text(angle = -45, hjust = 0, vjust=0.75)) + scale_color_manual(values=ap) + theme(text = element_text(size=16))
noLegendsigplot <- sigplot +theme(legend.position = "none")
noLegendsigplot
```

```{r}
#ggsave("sigplotnoLegend.pdf", height = 4, width = 6)
```

```{r, warning = FALSE}
sigplot
```
```{r}
#ggsave("sigplot.pdf")
```

# Session Info
```{r}
sessionInfo()
```

